### PR TITLE
Fix issue #2458

### PR DESCRIPTION
**Pull request by MatthewStadter**

### CODE REVIEW

File: lib/models/io/attachment.dart [hunk 1]
```diff
@@ -223,7 +223,7 @@ class Attachment {
 
   String get directory => "$baseDirectory/$guid";
 
-  String get path => "$directory/$transferName";
+  String get path => Platform.isWindows ? "$directory/$transferName".replaceAll(RegExp(r'[<>:"/\|?*]'), "_") : "$directory/$transferName";
```
Comment (jjoelj) [inline lib/models/io/attachment.dart:226]: - This changes the file structure, which I assume you didn't intend to do.
- Make sure the original path separators are maintained. (i.e. you should only run replace on the `transferName`, or we'll end up with a flat file structure.)
- We should also filter out forward slashes in filenames on Linux for completeness's sake.

**Review by jjoelj — CHANGES_REQUESTED** [review]
Thanks for contributing!

File: lib/models/io/attachment.dart (hunk 1 above)
Comment (MatthewStadter) [reply lib/models/io/attachment.dart]: Hello, I fixed the issues. Thanks

**Comment by tneotia** [comment]
@jjoelj we can do that ourselves :)

**Review by tneotia — APPROVED** [review]
LGTM, thanks for your contribution!

**Review by jjoelj — APPROVED** [review]

**Pull request merged**